### PR TITLE
[TTAHUB-3281] manual mark goals similar

### PR DESCRIPTION
**Pull request by GarrettEHill**
## Description of change

Implemented the ability to manually mark goals as similar within the Head Start TTA Hub. This new feature allows users to select multiple goals and group them together if they are determined to be similar. The change includes updates to the frontend UI to support this functionality, new routes, and backend modifications to process and store the similarity groupings.

## How to test

1. Navigate to the goals management section of the application.
2. Select multiple goals to mark them as similar.
3. Verify that the selected goals are grouped correctly.
4. Ensure the backend processes and stores these groupings as expected.
5. Run the updated unit tests to confirm all are passing.

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-3281

### Every PR

- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [x] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Before merge to main

- [ ] OHS demo complete
- [ ] Ready to create production PR

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: src/routes/recipient/index.js
Comment (nvms) [inline src/routes/recipient/index.js:47]: Let's avoid route paths that require case sensitivity if possible

Comment (GarrettEHill) [reply src/routes/recipient/index.js:47]:
```suggestion
  '/:recipientId/mark-similar-goals',
```

**Review by nvms — APPROVED** [review]
lgtm, but let's get another approval before merging

File: frontend/src/components/GoalCards/GoalsCardsHeader.js
Comment (AdamAdHocTeam) [inline frontend/src/components/GoalCards/GoalsCardsHeader.js:84]: Is an empty [] on use effect allowed by our linter rules?

Comment (nvms) [reply frontend/src/components/GoalCards/GoalsCardsHeader.js:84]: useEffect with empty deps is standard, and means "run only once" - what's the concern?

File: frontend/src/components/GoalCards/GoalsCardsHeader.js
Comment (AdamAdHocTeam) [inline frontend/src/components/GoalCards/GoalsCardsHeader.js:123]: Why do we trigger a manual reload of the page?

File: frontend/src/components/GoalCards/GoalsCardsHeader.js
Comment (AdamAdHocTeam) [reply frontend/src/components/GoalCards/GoalsCardsHeader.js:84]: NVM seems like its a one and done thing.

File: frontend/src/components/GoalCards/GoalsCardsHeader.js
Comment (GarrettEHill) [reply frontend/src/components/GoalCards/GoalsCardsHeader.js:123]: do the newly added similarity group shows up in the banner

Comment (AdamAdHocTeam) [reply frontend/src/components/GoalCards/GoalsCardsHeader.js:123]: Shouldn't this be using state to trigger re render?

Comment (GarrettEHill) [reply frontend/src/components/GoalCards/GoalsCardsHeader.js:123]: I guess I dont know what you mean

Comment (GarrettEHill) [reply frontend/src/components/GoalCards/GoalsCardsHeader.js:123]: If you have an alternative, that would work better, please suggest the code change.

Comment (AdamAdHocTeam) [reply frontend/src/components/GoalCards/GoalsCardsHeader.js:123]: Are you reloading the page to display something that was not previously there? If yes its means you are changing the state of the page. In this case I would think you would want to use some sort of useState() var to trigger the re render and display the state change.

Comment (AdamAdHocTeam) [reply frontend/src/components/GoalCards/GoalsCardsHeader.js]: Pushed suggested change.

File: src/services/goalSimilarityGroup.ts
Comment (AdamAdHocTeam) [inline src/services/goalSimilarityGroup.ts:260]: Looks good assuming we are cleaning up any groups that might already have these goals.

**Review by AdamAdHocTeam — APPROVED** [review]
Looks good!

File: frontend/src/components/GoalCards/GoalsCardsHeader.js
Comment (nvms) [inline frontend/src/components/GoalCards/GoalsCardsHeader.js:54]: Probably don't need this

Comment (GarrettEHill) [reply frontend/src/components/GoalCards/GoalsCardsHeader.js:54]:
```suggestion
```

**Pull request merged**
